Mod2/Lessons/Week2/DataRelationships.md

```diff
@@ -0,0 +1,158 @@
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_1.png'>
+</p>
+
+In our database, we have a table that looks similar to an excel sheet where each row of this table is one song entry or record in our database. So far, this is a fairly simple database, holding on to only a few attributes.  Without much difficulty, we could update our database to hold additional song attributes by adding additional columns to our songs table.
+
+But, what if we want to keep track of the artist that wrote each song?  We _could_ add a column to our database called `artist` and store the artist directly on a song, like this:
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_2.png'>
+</p>
+
+Seems like it works, right? Well... not exactly.  What if we wanted to also store information about specific artists, like their hometown or years active.  If we imagine what that might look like, we would see more columns on our `songs` table and many of those columns would be storing the exact same information - every Prince song would be storing 'Minneapolis' and '1975-2016':
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_3.png'>
+</p>
+
+This is very repetitive!
+
+## Normalizing our Database
+
+In order to cut down on redundant data on a specific table in our database, we can rely on **normalization**.  Normalization is a process by which we break down tables in our database so that each table relates only to one resource.  In the example we are working with, when we look at our songs table, we are storing information about songs and about artists. Really, we are trying to store information about two resources on one table - let's fix that.
+
+In your pairs, draw how you would normalize this database, so that songs and artists each have their own table. I highly recommend you look back at the Intro to Databases lesson to see how we normalized customers and orders!
+
+When we get back, be prepared to share your diagrams and answer the following questions.
+
+* How are you able to still know which artist wrote which song?
+* What primary key do you have?
+* What foreign key do you have?
+
+<details><summary>Normalized Songs and Artists Database</summary><br/>
+
+When we normalize our database, we need to include some sort of marker for ourselves to know how to relate the tables to one another.  We do this with **primary keys** and **foreign keys**.  As our database exists now, every record has an `id`.  We can use these ids to relate information in one table, to information in the other (or give each of our songs an artist).  For each song in our database, we will add an attribute `artist_id` that will hold the `id` of the artist that song belongs to.
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_5.png'>
```

Comment:
My least favorite thing in the world, but it's gonna happen...
"What are Talking Heads?" or sadly "Which Prince are we talking about?" 😭 
Because we're teaching new students, and 1991 was over 30 years ago 🙀 , how do you feel about newer references? OR should we use the older artists to teach these @#$% kids? 🤣